tempto-core/src/main/java/com/teradata/tempto/internal/query/CassandraQueryExecutor.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.teradata.tempto.internal.query;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.ColumnDefinitions;
+import com.datastax.driver.core.ColumnMetadata;
+import com.datastax.driver.core.DataType;
+import com.datastax.driver.core.KeyspaceMetadata;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.TableMetadata;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
+import com.teradata.tempto.configuration.Configuration;
+import com.teradata.tempto.query.QueryExecutionException;
+import com.teradata.tempto.query.QueryResult;
+
+import java.sql.JDBCType;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.Lists.newArrayList;
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+
+public class CassandraQueryExecutor
+    implements AutoCloseable
+{
+    private static final Map<DataType, JDBCType> typeMapping;
+    private final Cluster cluster;
+    private Session session;
+
+    static {
+        typeMapping = ImmutableMap.<DataType, JDBCType>builder()
+                .put(DataType.ascii(), JDBCType.VARCHAR)
+                .put(DataType.bigint(), JDBCType.BIGINT)
+                .put(DataType.blob(), JDBCType.BLOB)
+                .put(DataType.cboolean(), JDBCType.BOOLEAN)
+                .put(DataType.counter(), JDBCType.BIGINT)
+                .put(DataType.date(), JDBCType.DATE)
+                .put(DataType.decimal(), JDBCType.DECIMAL)
+                .put(DataType.cdouble(), JDBCType.DOUBLE)
+                .put(DataType.cfloat(), JDBCType.REAL)
+                .put(DataType.cint(), JDBCType.INTEGER)
+                .put(DataType.smallint(), JDBCType.SMALLINT)
+                //.put(DataType.text(), JDBCType.NVARCHAR)
```

Comment:
There's no SET/MAP/LIST type in JDBC. Presto treats them as "clob"/"varchar" but it's not universal, so I believe we shouldn't add them unless we need them.
Seems like there's no universal method to convert Cassandra types to JDBC types. In Presto's `CassandraMetadata` we have mapping hardcoded.
I don't believe we should blindly follow how we map types there because e.g. MAP -> VARCHAR mapping is valid for Presto only. It's not universally correct.

Moreover, I don't think this mapping is super important now because the only use of `CassandraQueryExecutor.executeQuery` is to create Cassandra tables. So the output (where JDBC translation happens) is not used at all for now.

When we notice we need to expose `executeQuery` to the test case (e.g. expose `executeOnCassandra` method), we'll have more issues than just how to map non-standard types.

I vote for postponing resolution of this problem.